pkg/cloudprovider/aws/launchtemplate.go

```diff
@@ -216,7 +216,8 @@ func (p *LaunchTemplateProvider) blockDeviceMappings(blockDeviceMappings []*v1al
 	blockDeviceMappingsRequest := []*ec2.LaunchTemplateBlockDeviceMappingRequest{}
 	for _, blockDeviceMapping := range blockDeviceMappings {
 		blockDeviceMappingsRequest = append(blockDeviceMappingsRequest, &ec2.LaunchTemplateBlockDeviceMappingRequest{
-			DeviceName: blockDeviceMapping.DeviceName,
+			DeviceName:  blockDeviceMapping.DeviceName,
+			VirtualName: blockDeviceMapping.VirtualName,
 			Ebs: &ec2.LaunchTemplateEbsBlockDeviceRequest{
 				DeleteOnTermination: blockDeviceMapping.EBS.DeleteOnTermination,
```

Comment:
These should be empty when supplying VirtualName, so they will panic on nill dereference here.

Reply:
I didn't get this one, @bwagner5 . Can you help?

Reply:
I think he means that `Ebs` should be set to `nil` if `VirtualName` is specified, so I guess something like

```go
		var ebs *ec2.LaunchTemplateEbsBlockDeviceRequest
		if blockDeviceMapping.VirtualName == nil {
			ebs = &ec2.LaunchTemplateEbsBlockDeviceRequest{
				DeleteOnTermination: blockDeviceMapping.EBS.DeleteOnTermination,
				Encrypted:           blockDeviceMapping.EBS.Encrypted,
				VolumeType:          blockDeviceMapping.EBS.VolumeType,
				Iops:                blockDeviceMapping.EBS.IOPS,
				Throughput:          blockDeviceMapping.EBS.Throughput,
				KmsKeyId:            blockDeviceMapping.EBS.KMSKeyID,
				SnapshotId:          blockDeviceMapping.EBS.SnapshotID,
				VolumeSize:          p.volumeSize(blockDeviceMapping.EBS.VolumeSize),
			}
		}

		blockDeviceMappingsRequest = append(blockDeviceMappingsRequest, &ec2.LaunchTemplateBlockDeviceMappingRequest{
			DeviceName:  blockDeviceMapping.DeviceName,
			VirtualName: blockDeviceMapping.VirtualName,
			Ebs:         ebs,
		})
```